### PR TITLE
[v9] Add missing DLPack test for complex numbers

### PR DESCRIPTION
**Pull request by leofang**
Manually backport part of #5306.

### CODE REVIEW

**Comment by leofang** [comment]
Jenkins, test this please

File: tests/cupy_tests/core_tests/test_dlpack.py [hunk 1]
```diff
@@ -20,6 +20,11 @@ def test_conversion(self, dtype):
         elif cupy.issubdtype(self.dtype, cupy.floating):
             self.array = cupy.random.rand(
                 2, 3).astype(self.dtype)
+        elif cupy.issubdtype(dtype, cupy.complexfloating):
```
Comment (kmaehashi) [inline tests/cupy_tests/core_tests/test_dlpack.py:23]: Nit for consistency :smile:
```suggestion
        elif cupy.issubdtype(self.dtype, cupy.complexfloating):
```

**Comment by kmaehashi** [comment]
pfnCI, test this please.

File: tests/cupy_tests/core_tests/test_dlpack.py (hunk 1 above)
Comment (leofang) [reply tests/cupy_tests/core_tests/test_dlpack.py]: Good catch! lol

**Review by kmaehashi — APPROVED** [review]

**Comment by kmaehashi** [comment]
LGTM! Test failures are not related to this PR.

**Pull request merged**